File: frontend/src/types/dto/briefingDto.ts
// briefingDto.ts - DTOs Briefing
// Miroir de Orion.Core/DTOs/Responses/BriefingDto.cs — garder les deux alignés.
//
// `shiftStarStats`, `calendarEvents`, `unreadEmails` et `summary` ont été retirés : le serveur
// ne les a jamais envoyés. L'overlay affichait `summary`, donc du vide, depuis toujours.

/** Un article réellement collecté : c'est ce qui rend le briefing vérifiable. */
export interface BriefingSource {
  title: string;
  url: string;
  source: string;
  /** 'local' | 'africa' | 'world' */
  circle: string;
}

export interface BriefingDto {
  id: string;
  /** Le texte du briefing — c'est CE champ que le serveur remplit. */
  content: string;
  createdAt: string;
  stats?: Record<string, unknown>;
  sources?: BriefingSource[];
}

File: backend/Orion.Core/DTOs/Responses/BriefingDto.cs
namespace Orion.Core.DTOs.Responses;

public class BriefingDto
{
    public Guid Id { get; set; }
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public Dictionary<string, object>? Stats { get; set; }

    /// <summary>
    /// Les articles reellement collectes. Toute la garantie anti-invention repose sur le fait
    /// que chaque sujet du briefing vient d'une URL : ne pas les renvoyer rendrait cette
    /// garantie invisible, donc invérifiable.
    /// </summary>
    public List<BriefingSource> Sources { get; set; } = new();
}

public class BriefingSource
{
    public string Title { get; set; } = string.Empty;
    public string Url { get; set; } = string.Empty;
    public string Source { get; set; } = string.Empty;

    /// <summary>« local », « africa » ou « world ».</summary>
    public string Circle { get; set; } = string.Empty;
}
